cumulus_library/protected_table_builder.py

```diff
@@ -37,21 +37,30 @@ def prepare_queries(
         study_name: str,
         study_stats: dict,
         *args,
+        manifest: study_parser.StudyManifestParser | None = None,
         **kwargs,
     ):
+        if manifest and manifest.get_dedicated_schema():
+            db_schema = manifest.get_dedicated_schema()
```

Comment:
Why isn't the `get_schema()` check in cli.py not saving you from having to check this flag downstream in order to use the right schema?

I get why you still need some logic for the table names -- though... if we saved somewhere a field like `table_namespace = f"{study_prefix}__"` and set that to `""` when in a dedicated schema study, we could reduce logic checks around the place - which feel a little brittle - easy to forget a place where we need custom logic for this case. Just feels like this feature *could* be, if we were set up for it, a single block of like:
```
if dedicated:
  create_db()
  schema = dedicated
  short_prefix = ""
else:
  schema = cli_schema
  short_prefix = f"{study_prefix}__"
full_prefix = f"{schema}.{short_prefix}"
```
or something.

Reply:
this is some of what drove the 'this is the last breaking change' comment. If the schema moved into the config, this all becomes much simpler.

my instinct is to sort of live with this until biting that off - noting this for #225